pyccel/codegen/printing/ccode.py

```diff
@@ -1208,6 +1209,34 @@ def _print_NumpyRandint(self, expr):
     def _print_NumpyMod(self, expr):
         return self._print(PyccelMod(*expr.args))
 
+    def _print_NumpyLinspace(self, expr):
+        template = '({start} + {index}*{step})'
+        if not isinstance(expr.endpoint, LiteralFalse):
+            template = '({start} + {index}*{step})'
+            lhs = self._print(expr.get_user_nodes(Assign)[0].lhs)
+            lhs = lhs.replace(self._print(expr.ind), self._print(PyccelMinus(expr.num, LiteralInteger(1), simplify = True)))
```

Comment:
If you need something safer I would suggest something like:
```
lhs = expr.get_user_nodes(Assign)[0].lhs
lhs.substitute(expr.ind, PyccelMinus(expr.num, LiteralInteger(1), simplify = True))
lhs_code = self._print(lhs)
```
(this will only work once the discussion above concerning `_attribute_nodes` is fixed)

Reply:
updated